src/LinearMaps.jl

```diff
@@ -69,6 +73,21 @@ convert_to_lmaps(A) = (convert_to_lmaps_(A),)
 @inline convert_to_lmaps(A, B, Cs...) =
     (convert_to_lmaps_(A), convert_to_lmaps_(B), convert_to_lmaps(Cs...)...)
 
+_front(As::Tuple) = Base.front(As)
+_front(As::AbstractVector) = @inbounds @views As[1:end-1]
+_tail(As::Tuple) = Base.tail(As)
+_tail(As::AbstractVector) = @inbounds @views As[2:end]
+
+_combine(A::LinearMap, B::LinearMap) = tuple(A, B)
+_combine(A::LinearMap, Bs::LinearMapTuple) = tuple(A, Bs...)
+_combine(As::LinearMapTuple, B::LinearMap) = tuple(As..., B)
+_combine(As::LinearMapTuple, Bs::LinearMapTuple) = tuple(As..., Bs...)
+_combine(A::LinearMap, Bs::LinearMapVector) = vcat(A, Bs...)
```

Comment:
I don't think you need to splat vector arguments in `vcat`. This could lead to calling `vcat` with a large number of arguments, and thus trigger additional compilation.

Reply:
Yes, of course, that was oversight!

Reply:
I originally meant to not splat, so this was oversight. But now I benchmarked, and it turns out that splatting has a very positive effect for both short (5)  and long (500) vectors. For instance:

```julia
julia> @btime vcat(x..., y...) setup=(x = (rand(500)...,); y = rand(500));
  47.389 μs (1003 allocations: 35.50 KiB)

julia> @btime vcat(x..., y) setup=(x = (rand(500)...,); y = rand(500))
  11.418 ms (259012 allocations: 5.98 MiB)

julia> @btime vcat(x..., y...) setup=(x = (rand(5)...,); y = rand(5))
  345.180 ns (12 allocations: 352 bytes)

julia> @btime vcat(x..., y) setup=(x = (rand(5)...,); y = rand(5))
  1.230 μs (37 allocations: 944 bytes)
```

Reply:
Ha, it turns out that this was the wrong approach anyway. We need `Base.vect` instead of `vcat`, because the latter constructs a `BlockMap`.